docs/source/deployment/tgi.rst

```diff
@@ -0,0 +1,106 @@
+TGI
+=====================
+
+Hugging Face's Text Generation Inference (TGI) is a production-ready framework specifically designed for deploying and serving large language models (LLMs) for text generation tasks. It offers a seamless deployment experience, powered by a robust set of features:
+
+* **Speculative Decoding:**  Accelerates generation speeds.
+* **Tensor Parallelism:**  Enables efficient deployment across multiple GPUs.
+* **Token Streaming:**  Allows for the continuous generation of text.
+* **Versatile Device Support:**  Works seamlessly with AMD, Gaudi, AWS Inferentia, and Google TPU.
+
+
+Installation
+-----------------
+
+The easiest way to use TGI is via the TGI docker image, but it's possible to run it locally via Conda. Please refer this doc for instructions on the installation: `https://github.com/huggingface/text-generation-inference?tab=readme-ov-file#local-install`
+
+
+Deploy Qwen with TGI
+-----------------------
+
+Find a model from qwen2 collection and get the model path: `https://huggingface.co/collections/Qwen/qwen2-6659360b33528ced941e557f`
+
+You can either run the model via CLI or via Docker
+
+
+.. code:: bash
+
+   # CLI
+   text-generation-launcher --model-id Qwen/Qwen2-0.5B-Instruct --port 8080
+
+   # via Docker
+   model=Qwen/Qwen2-0.5B-Instruct
+   volume=$PWD/data # share a volume with the Docker container to avoid downloading weights every run
+
+   docker run --gpus all --shm-size 1g -p 8080:80 -v $volume:/data ghcr.io/huggingface/text-generation-inference:2.0 --model-id $model
+
+
+Using TGI API
+-------------
+
+Once deployed, the model will be available on the selected port
+
+.. code:: bash
+
+   curl 127.0.0.1:8080/generate_stream \
+           -X POST \
+           -d '{"inputs":"What is Deep Learning?","parameters":{"max_new_tokens":20}}' \
+           -H 'Content-Type: application/json'
+
+It's also available on OpenAI style API.
+
+.. code:: bash
+
+   curl http://localhost:8080/v1/chat/completions -H "Content-Type: application/json" -d '{
+           "model": "Qwen/Qwen2-0.5B-Instruct",
+           "messages": [
+           {"role": "system", "content": "You are a helpful assistant."},
+           {"role": "user", "content": "Tell me something about large language models."}
+           ],
+           }'
+
+The full API is listed in this Swagger UI: `https://huggingface.github.io/text-generation-inference/#/Text%20Generation%20Inference/completions`
```

Comment:
that page has a huge error toast at the top

![image](https://github.com/QwenLM/Qwen2/assets/17811943/6a161a75-0e85-45a8-975b-6ffff197c281)